### DISCUSSION TITLE
Creating a repo from a template using an installation or user token

### DISCUSSION THREAD

**Discussion by Styren** [community/community#52316] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi, I can't for the life of me figure out how to create a repo on behalf of another user using a templated repo. I keep getting this error:
```json

{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/repos#create-a-repository-using-a-template"
}
```

Creating a normal repo works fine, but when I try to do so using a [publicly accessible template repo](https://github.com/symbiosis-cloud/k8s-microservice-template) I get the error above. I've tried both with an installation access token and with a user token with the same results.

Forking the repo using the API works fine, but generating from a template does not.

Is it even possible to create a repo using a template on behalf of a user? Is this a bug?

Permissions:
Read access to code and metadata
Read and write access to administration, checks, and pull requests

**Comment by SMelidoni** [comment, marked as answer]
Hi @Styren 👋🏽

Unfortunately, you cannot create a repository using a template on behalf of another user through the API. As per the [GitHub API Documentation](https://docs.github.com/en/rest/reference/repos#create-a-repository-using-a-template), you can only create a repository using a template for the authenticated user or an organization they belong to.

The error message you're receiving, "Resource not accessible by integration," indicates that the operation you're trying to perform is not allowed for the type of token you're using.

As a workaround, you can:

1. Fork the template repository on behalf of the user (as you mentioned that this works for you).
2. Apply any necessary changes or customizations to the forked repository programmatically.
3. Instruct the user to change the forked repository's settings to use it as a template, if needed.

Please note that this is not an ideal solution, but it's a workaround due to the limitations of the GitHub API.

I hope this helps clarify the situation. If you have any further questions or need assistance, please let me know.

**Reply by nveeramn** [reply]
@SMelidoni I'm seeing a 404 template not issue when creating a repo from a template using a GitHub App token, Is this a known issue?

**Reply by lotsathought** [reply]
- [ ]

**Reply by lotsathought** [reply]
Yes it's recurring

**Reply by naturedamends** [reply]
Why does the docs say this then:

```
[Fine-grained access tokens for "Create a repository using a template"](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-using-a-template--fine-grained-access-tokens)
This endpoint works with the following fine-grained token types:

[GitHub App user access tokens](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-user-access-token-for-a-github-app)
[GitHub App installation access tokens](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-an-installation-access-token-for-a-github-app)
[Fine-grained personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token)
```

I even own the template
